subsys/bootloader/cmake/sign.cmake

```diff
@@ -4,61 +4,75 @@
 # SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
 #
 
-set(sign_depends kernel_elf ${KEY_FILE_DEPENDS})
 
-set(hash_file firmware.sha256)
-set(signature_file firmware.signature)
-set(SIGNATURE_PUBLIC_KEY_FILE public.pem)
+#TODO: Detect what kernel_elf to depend on and resolve what
+#      image should be hashed and signed by b0.
+
+set(KERNEL_ELF_TO_BE_SIGNED kernel_elf)
+
+set(GENERATED_FILES_PATH ${PROJECT_BINARY_DIR}/nrf/subsys/bootloader/generated)
+# This is needed for make, ninja is able to resolve this but make is not
+add_custom_target(
+  generated_file_directory_target
+  COMMAND
+  ${CMAKE_COMMAND} -E make_directory ${GENERATED_FILES_PATH}
```

Comment:
Usually we do this at Configure-time.

file(MAKE_DIRECTORY ${PROJECT_BINARY_DIR}/include/generated)

Any reason why we are doing it at build-time here?

Reply:
It fails when using make apparently.

Reply:
Tried it again today seems there probably was a dependency fix yesterday that fixed this. It should work now and will switch it.